dotnet/src/SemanticKernel/Planning/FunctionFlowRunner.cs

```diff
@@ -154,12 +168,56 @@ private async Task<string> ProcessNodeListAsync(XmlNodeList nodeList, string fun
                     continue;
                 }
 
+                // Ignoring first ELSE after the If that was already processed
+                if (ignoreElse && o2.Name.StartsWith(ConditionElseTag, StringComparison.InvariantCultureIgnoreCase))
+                {
+                    context.Log.LogTrace("{0}: ignoring ELSE tag node", parentNodeName);
+                    ignoreElse = false;
+                    continue;
+                }
+
+                if (processFunctions && o2.Name.StartsWith(ConditionIfTag, StringComparison.InvariantCultureIgnoreCase))
```

Comment:
Can you add unit tests for code coverage here?

Reply:
Added the unit tests for the different Planner scenario cases.